### PR TITLE
Get reads from sv

### PR DESCRIPTION
**Pull request by adrosenbaum**
**background**

Recently, a tandem duplication spiked into a background set by mutacc failed to get detected be called in mip, as TIDDIT did not find sufficient support for it. Mutacc padds each genomic region around a causative variant when extracting the reads, however some SVs might require a larger padding to be detected. This PR allows the user to specify a separate padding for SVs, which might fix this problem. The default padding for SVs will now be 1 kbp.

- Allows user to pass different paddings to be used for SVs and SNVs/indels respectively

- Paddings can now be specified in config file, or in CLI.

- Paddings will be adjusted with regards to read_length. For SVs, the padding will also get adjusted with regards to the CIPOS and CIEND flags if present in the vcf.

- The code in mutacc/utils/bam_handler.py has been refactored, moving all functions dealing with a bam file to the BAMContext context manager.

- The data for the demo case has been modified to also include a duplication, artificially inserting a DUP in a NIST case.  

**How to test**:
1. Install this branch on stage
1. run ```cg upload process-solved <case_id>``` for some cases having SNVs and SVs as marked causatives

**Expected outcome**:

1. Inspect the extracted reads. The number of reads extracted from the SV cases should be larger than before (This is because a greater padding will be used for SVs)

**Review:**
- [x] code approved by @sylvinite @henrikstranneheim 
- [x] tests executed by @adrosenbaum 
- [x] "Merge and deploy" approved by @adrosenbaum 
Thanks for filling in who performed the code review and the test!

This is minor **version bump** because it adds functionality in a backwards compatible manner

### CODE REVIEW

**Comment by adrosenbaum** [comment]
@sylvinite here is a first PR for you. If you want we can go through it together. Feel free to pass on any PR from microsalt that you need to get reviewed.

@henrikstranneheim this is the problem we discussed some week ago, where mip failed to call a  SV from a mutacc dataset. I think this might fix the problem.

File: mutacc/utils/bam_handler.py [hunk 1]
```diff
@@ -198,84 +202,51 @@ def make_names_temp(self, out_dir):
...
-    paired = False
-    with pysam.AlignmentFile(bam_file) as bam:
+        Returns:
+            paired (bool): True if reads are paired
+        """
+
+        insert_sizes = []
+        acc_length = 0
+        paired = False
         count = 0
```
Comment (henrikstranneheim) [inline mutacc/utils/bam_handler.py]:
```suggestion
        read_count = 0
```

File: mutacc/utils/bam_handler.py [hunk 2]
```diff
@@ -198,84 +202,51 @@ def make_names_temp(self, out_dir):
...
-        for read in bam:
+        no_reads = 10000
```
Comment (henrikstranneheim) [inline mutacc/utils/bam_handler.py]:
```suggestion
        MAX_READ_COUNT = 10000
```
It is a constant after all 😄

File: mutacc/builds/build_variant.py [hunk 3]
```diff
@@ -18,19 +18,40 @@ class Variant(dict):
...
+        del self.entry
+
+    def _get_padding(self, variant_type):
+
+        if variant_type == "SNV":
+            return self.padding
+
+        if self.entry.INFO.get("CIPOS") or self.entry.INFO.get("CIEND"):
+
+            cis = []
```
Comment (henrikstranneheim) [inline mutacc/builds/build_variant.py]: What is cis in this context?

File: mutacc/builds/build_variant.py [hunk 4]
```diff
@@ -18,19 +18,40 @@ class Variant(dict):
...
+            if self.entry.INFO.get("CIPOS"):
+                cis.append(abs(self.entry.INFO["CIPOS"][0]))
+            if self.entry.INFO.get("CIEND"):
+                cis.append(abs(self.entry.INFO["CIEND"][-1]))
+            ci_max = max(cis)
```
Comment (henrikstranneheim) [inline mutacc/builds/build_variant.py]: Is this confidence interval?

File: mutacc/builds/build_case.py [hunk 5]
```diff
@@ -69,7 +69,7 @@ def _build_variants(self, padding=None, vcf_parse=None):
         variant_objects = []
 
         for variant_object in get_variants(
-            self.input_case["variants"], padding=padding, vcf_parse=vcf_parse
+            self.input_case["variants"], padding=padding, sv_padding=sv_padding, vcf_parse=vcf_parse
```
Comment (sylvinite) [inline mutacc/builds/build_case.py]: Consider keywording the first param for the function as well

File: mutacc/builds/build_variant.py [hunk 6]
```diff
@@ -18,19 +18,40 @@ class Variant(dict):
...
             self.update(parser.parse(vcf_entry))
 
+        del self.entry
+
+    def _get_padding(self, variant_type):
+
+        if variant_type == "SNV":
+            return self.padding
+
+        if self.entry.INFO.get("CIPOS") or self.entry.INFO.get("CIEND"):
```
Comment (sylvinite) [inline mutacc/builds/build_variant.py]: Consider if-else block

File: mutacc/cli/constants.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    #      |   |     fastq_files
+    #      |   sample2
+    #      |   |     |
+    #      |   |     fastq_files
+    #      |   sample2
+    #      |         |
+    #      |         fastq_files
+    #      case2
+    #      ...
+    'read_dir': 'reads',
```
Comment (sylvinite) [inline mutacc/cli/constants.py]: Consider using Path() function or similar for extended clarity.

File: mutacc/builds/build_variant.py (hunk 3 above)
Comment (adrosenbaum) [reply mutacc/builds/build_variant.py]: confidence intervals 😄 will rename variable to something nicer.

File: mutacc/builds/build_variant.py (hunk 4 above)
Comment (adrosenbaum) [reply mutacc/builds/build_variant.py]: yes, CIPOS and CIEND are given by some SV callers, giving a confidence interval e.g. CIPOS=-100,100;CIEND=-50,50 around the start and beginning. I wanted to incorporate this into the padding for the SVs that have these flags present.

File: mutacc/builds/build_case.py (hunk 5 above)
Comment (adrosenbaum) [reply mutacc/builds/build_case.py]: will do!

File: mutacc/builds/build_variant.py (hunk 6 above)
Comment (adrosenbaum) [reply mutacc/builds/build_variant.py]: I think you are right

**Pull request merged**